spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -552,6 +557,30 @@ public void setUseTemporaryReplyQueues(boolean value) {
 	}
 
 	/**
+	 * Set an expression to be evaluated to set the userId message property if it
+	 * evaluates to a non-null value and the property is not already set in the
+	 * message to be sent.
+	 * See https://www.rabbitmq.com/validated-user-id.html
+	 * @param userIdExpression the expression.
+	 * @since 1.6
+	 */
+	public void setUserIdExpression(Expression userIdExpression) {
+		this.userIdExpression = userIdExpression;
+	}
+
+	/**
+	 * Set an expression to be evaluated to set the userId message property if it
+	 * evaluates to a non-null value and the property is not already set in the
+	 * message to be sent.
+	 * See https://www.rabbitmq.com/validated-user-id.html
+	 * @param userIdExpression the expression.
```

Comment:
There is no such a method parameter.
I think IDEA is good with this inspection :smile: 
Will rename during merge, if only concern.